File: Roman_to_Integer/Roman_to_Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char, int> romanMap = {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };
        int ans=0;
        for(int i=0;i<s.size();i++){
            if(romanMap[s[i]] < romanMap[s[i+1]]){
                ans -= romanMap[s[i]];
            }else{
                ans += romanMap[s[i]];
            }
        }
        return ans;
    }
};

File: Roman_to_Integer/Roman_to_Integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        dic = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        i = 0
        ans = 0;
        previous = 0
        for i in range(len(s)):
            if dic[s[i]]>previous:
                ans = ans - 2*previous + dic[s[i]]
            else:
                ans += dic[s[i]]
            previous = dic[s[i]]
        return ans
